src/features/serde/de_borrowed.rs

```diff
@@ -0,0 +1,401 @@
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_i32(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_i64<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_i64(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_u8<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_u8(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_u16<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_u16(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_u32<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_u32(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_u64<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_u64(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_f32<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_f32(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_f64<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_f64(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_char<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_char(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_str<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        let str = <&'de str>::borrow_decode(&mut self.de)?;
+        visitor.visit_borrowed_str(str)
+    }
+
+    fn deserialize_string<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_string(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_bytes<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        let bytes = <&'de [u8]>::borrow_decode(&mut self.de)?;
+        visitor.visit_borrowed_bytes(bytes)
+    }
+
+    fn deserialize_byte_buf<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        visitor.visit_byte_buf(Decode::decode(&mut self.de)?)
+    }
+
+    fn deserialize_option<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: serde_incl::de::Visitor<'de>,
+    {
+        let val = Option::<()>::decode(&mut self.de)?;
```

Comment:
Does this really only get called for `Option<()>`?

Reply:
I'm using `Option<()>` because I only care about the `Some`/`None` variant, and I know `()` won't actually read from the stream.
Alternatively I could ready `u8` here, but then we have multiple places where we're decoding an `Option`.